src/factories/custom/Restricted_PIM_Factory_v1.sol

```diff
@@ -30,6 +32,8 @@ import {MintWrapper} from "src/external/token/MintWrapper.sol";
 
 // External Dependencies
 import {ERC2771Context, Context} from "@oz/metatx/ERC2771Context.sol";
+import {ERC165Upgradeable} from
```

Comment:
Any reason we use the `Upgradeable` variant here? We only use it to check for a supported interface, so the version doesn't matter really, but as this contract isn't upgradeable to me it seems like we should use the regular one here too?